packages/abi-coder/test/Interface.test.ts

```diff
@@ -766,4 +766,42 @@ describe('Abi interface', () => {
       );
     });
   });
+
+  describe('decodeLog', () => {
+    it('should return decoded log by id', () => {
+      const data = exhaustiveExamplesInterface.decodeLog('0x01000000000000000000000000000020', '0');
+      expect(data).toEqual({
+        a: true,
+        b: 32,
+      });
+    });
+
+    it('should throw an error when log does not exist', () => {
+      expect(() =>
+        exhaustiveExamplesInterface.decodeLog('0x01000000000000000000000000000020', '1')
+      ).toThrowError(`Log type with logId '1' doesn't exist in the ABI.`);
```

Comment:
You should use the `expectToThrowFuelError` utility here ([example](https://github.com/FuelLabs/fuels-ts/blob/ba0f3d0f7001cd1f261d658bec0a356a1a40a4ee/packages/abi-coder/src/encoding/coders/ArrayCoder.test.ts#L93)).